### PR TITLE
fix: catch pp wrong due to both acc and judgements specified

### PR DESCRIPTION
**Pull request by minisbett**
# Describe your changes
PP breaks on catch if both the accuracy and the 50s are specified. In VC we argued the best way to solve this is raising an exception in the usecase, and dropping specifying acc where possible, since it's a useless info.

This PR fixes catch PP, meaning bancho.py hosters will need to re-calculate all PP. Before doing this, I'd suggest looking into #573, as it improves the functionality of the recalc script and might be significant with this change.
## Checklist
- [x] I've manually tested my code

### CODE REVIEW

File: app/usecases/performance.py [hunk 1]
```diff
@@ -65,16 +65,8 @@ def calculate_performances(
...
-        # if (score.acc is None) == (
-        #     score.n300 is None
-        #     and score.n100 is None
-        #     and score.n50 is None
-        #     and score.ngeki is None
-        #     and score.nkatu is None
-        #     and score.nmiss is None
-        # ):
-        #     raise ValueError("Either acc OR 300/100/50/geki/katu/miss must be present")
+        if score.acc and (score.ngeki or score.nkatu or score.n50 or score.n100):
```
Comment (cmyui) [inline app/usecases/performance.py:68]: n300?

**Comment by tsunyoku** [comment]
bug fixes like this are really good candidates for adding unit tests to cover imo

File: tools/recalc.py [hunk 2]
```diff
@@ -66,7 +66,6 @@ async def recalculate_score(
     calculator = Calculator(
```
Comment (cmyui) [inline tools/recalc.py:66]: can we have a similar check/err raise here?

Comment (cmyui) [reply tools/recalc.py:66]: or even better, could we make the recalc.py script use the usecase function?

Comment (tsunyoku) [reply tools/recalc.py:66]: this should probably be using the usecase, no?

Comment (minisbett) [reply tools/recalc.py:66]: > an we have a similar check/err raise here?
we don't need one, we simply just don't specify the acc here, why would we check for anything

File: app/usecases/performance.py (hunk 1 above)
Comment (minisbett) [reply app/usecases/performance.py:68]: When does it make sense to specify n300? With score.acc it's estimated, and when u specify 100s etc. n300 is just the leftover.

Comment (cmyui) [reply app/usecases/performance.py:68]: Yeah I agree - I think we should _block_ usage of `n300` and `acc` together, the same as with geki/katu/n50/n100.

More specifically, outside of misses, any fields which affect `acc`:

image

File: tools/recalc.py (hunk 2 above)
Comment (cmyui) [reply tools/recalc.py:66]: Oh I see, this should be using the usecase yeah

File: app/usecases/performance.py [hunk 3]
```diff
@@ -60,21 +61,24 @@ def calculate_performances(
...
-        #     and score.n50 is None
-        #     and score.ngeki is None
-        #     and score.nkatu is None
-        #     and score.nmiss is None
-        # ):
-        #     raise ValueError("Either acc OR 300/100/50/geki/katu/miss must be present")
+        if score.acc and (
+            score.n300 or score.n100 or score.n50 or score.ngeki or score.nkatu
+        ):
+            raise ValueError("Must not specify both accuracy and geki/katu/50s/100s.")
```
Comment (cmyui) [inline app/usecases/performance.py:81]:
```suggestion
            raise ValueError("Must not specify accuracy AND 300/100/50/geki/katu. Only one or the other.")
```

File: tools/recalc.py (hunk 2 above)
Comment (cmyui) [reply tools/recalc.py:66]: Follow-up PR is fine

**Review by cmyui — DISMISSED** [review]

File: app/usecases/performance.py (hunk 1 above)
Comment (minisbett) [reply app/usecases/performance.py]: I think we can just remove the n300 param completely? I dont think it ever has a usecase

**Review by cmyui — APPROVED** [review]

**Pull request merged**